common/djangoapps/student/views.py

```diff
@@ -708,7 +709,7 @@ def dashboard(request):
 
         for course_run in course_runs_for_course:
             enrollment_end = course_run.get('enrollment_end')
-            if not enrollment_end or enrollment_end > datetime.datetime.now(UTC):
+            if not enrollment_end or (dateutil.parser.parse(enrollment_end) > datetime.datetime.now(UTC)):
```

Comment:
Nit: would being explicit about enrollment_end be clearer than using the falsy'ness of none?
i.e.
if enrollment_end is None or (dateutil.parser.parse(enrollment_end) > datetime.datetime.now(UTC)):

Reply:
P.s. this is probably super style/nit,

Reply:
I don't think so, in this case. If the API returns an empty string instead of None for enrollment_end, I think we can still interpret that to mean that there isn't an enrollment end. Furthermore, passing a falsy value to parser.parse will likely break it.